################################################################################
SHORT DESCRIPTION: 
################################################################################
Build fixes for TBOOT in OpenXT.

################################################################################
LONG DESCRIPTION: 
################################################################################
Adds support for cross compiling in the OpenXT build and removes uneeded lib
dependencies in the lcp projects that cause build failures.

################################################################################
CHANGELOG 
################################################################################
Origins: Merged version of the 1.7.0 patches tboot-config-cross-compile.patch
and tboot-broken-lcptools-build.patch.

Intial Commit: Ross Philipson, philipsonr@ainfosec.com, 8/11/2015

################################################################################
REMOVAL 
################################################################################

################################################################################
UPSTREAM PLAN 
################################################################################
These are local OpenXT build changes, no upstream plan.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None

################################################################################
PATCHES 
################################################################################
Index: tboot-1.8.3/Config.mk
===================================================================
--- tboot-1.8.3.orig/Config.mk
+++ tboot-1.8.3/Config.mk
@@ -45,25 +45,25 @@ CFLAGS_WARN       = -Wall -Wformat-secur
 	            $(call cc-option,$(CC),-Wlogical-op,) \
 	            -Wno-missing-field-initializers
 
-AS         = as
-LD         = ld
-CC         = gcc
-CPP        = cpp
-AR         = ar
-RANLIB     = ranlib
-NM         = nm
-STRIP      = strip
-OBJCOPY    = objcopy
-OBJDUMP    = objdump
+AS         ?= as
+LD         ?= ld
+CC         ?= gcc
+CPP        ?= cpp
+AR         ?= ar
+RANLIB     ?= ranlib
+NM         ?= nm
+STRIP      ?= strip
+OBJCOPY    ?= objcopy
+OBJDUMP    ?= objdump
 
 ifeq ($(debug),n)
-INSTALL_STRIP = -s
+INSTALL_STRIP ?= -s
 endif
 
-INSTALL      = install
-INSTALL_DIR  = $(INSTALL) -d -m0755 -p
-INSTALL_DATA = $(INSTALL) -m0644 -p
-INSTALL_PROG = $(INSTALL) $(INSTALL_STRIP) -m0755 -p
+INSTALL      ?= install
+INSTALL_DIR  ?= $(INSTALL) -d -m0755 -p
+INSTALL_DATA ?= $(INSTALL) -m0644 -p
+INSTALL_PROG ?= $(INSTALL) $(INSTALL_STRIP) -m0755 -p
 
 
 #
Index: tboot-1.8.3/lcptools/Makefile
===================================================================
--- tboot-1.8.3.orig/lcptools/Makefile
+++ tboot-1.8.3/lcptools/Makefile
@@ -94,13 +94,13 @@ $(LCP2_LIB) : pol.o poldata.o pollist.o
 	$(AR) rc $@ $^
 
 lcp_crtpolelt : crtpolelt.o $(POLELT_PLUGINS) $(LCP2_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -llcp -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -o $@
 
 lcp_crtpollist : crtpollist.o $(POLELT_PLUGINS) $(LCP2_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -llcp -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -o $@
 
 lcp_crtpol2 : crtpol2.o $(POLELT_PLUGINS) $(LCP2_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -llcp -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -o $@
 
 lcp_mlehash : mlehash.o
 	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -lz -o $@
Index: tboot-1.8.3/lcptools-v2/Makefile
===================================================================
--- tboot-1.8.3.orig/lcptools-v2/Makefile
+++ tboot-1.8.3/lcptools-v2/Makefile
@@ -77,13 +77,13 @@ $(LCP2_LIB) : pol.o poldata.o pollist2.o
 	$(AR) rc $@ $^
 
 lcp2_crtpolelt : crtpolelt.o $(POLELT_PLUGINS) $(LCP2_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -llcp -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -o $@
 
 lcp2_crtpollist : crtpollist.o $(POLELT_PLUGINS) $(LCP2_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -llcp -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -o $@
 
 lcp2_crtpol : crtpol.o $(POLELT_PLUGINS) $(LCP2_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -llcp -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -o $@
 
 lcp2_mlehash : mlehash.o $(LCP2_LIB)
 	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lcrypto -lz -o $@
